presto-main/src/main/java/com/facebook/presto/sql/Serialization.java

```diff
@@ -107,7 +107,15 @@ public void serialize(VariableReferenceExpression value, JsonGenerator jsonGener
     public static class VariableReferenceExpressionDeserializer
             extends KeyDeserializer
     {
-        private final TypeManager typeManager;
+        // when it is on the test code path, jackson deserializer instantiates the default constructor with no arg,
```

Comment:
Is this paragraph still valid?

Reply:
The fix to avoid using `createTestFunctionAndTypeManager` in `Serialization` main code path is to create a json mapper provider using the SerDe function for VariableReferenceExpression and use this custom json codec for serialization. This does not need any special no-args handling on the testing code path, e.g. TestExplainVerification, the `planCodec` is bind with `JsonObjectMapperProvider` that uses `VariableReferenceExpressionDeserializer`.